homeassistant/components/nuki/lock.py

```diff
@@ -40,46 +35,37 @@
     }
 )
 
-LOCK_N_GO_SERVICE_SCHEMA = vol.Schema(
-    {
-        vol.Optional(ATTR_ENTITY_ID): cv.entity_ids,
-        vol.Optional(ATTR_UNLATCH, default=False): cv.boolean,
-    }
-)
-
 
-def setup_platform(hass, config, add_entities, discovery_info=None):
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
     """Set up the Nuki lock platform."""
-    bridge = NukiBridge(
-        config[CONF_HOST],
-        config[CONF_TOKEN],
-        config[CONF_PORT],
-        True,
-        DEFAULT_TIMEOUT,
-    )
 
-    devices = [NukiLockEntity(lock) for lock in bridge.locks]
+    def get_entities():
+        bridge = NukiBridge(
+            config[CONF_HOST],
+            config[CONF_TOKEN],
+            config[CONF_PORT],
+            True,
+            DEFAULT_TIMEOUT,
+        )
 
-    def service_handler(service):
-        """Service handler for nuki services."""
-        entity_ids = extract_entity_ids(hass, service)
-        unlatch = service.data[ATTR_UNLATCH]
+        entities = [NukiLockEntity(lock) for lock in bridge.locks]
+        entities.extend([NukiOpenerEntity(opener) for opener in bridge.openers])
+        return entities
 
-        for lock in devices:
-            if lock.entity_id not in entity_ids:
-                continue
-            lock.lock_n_go(unlatch=unlatch)
+    entities = await hass.async_add_executor_job(get_entities)
 
-    hass.services.register(
-        DOMAIN,
-        SERVICE_LOCK_N_GO,
-        service_handler,
-        schema=LOCK_N_GO_SERVICE_SCHEMA,
-    )
+    async_add_entities(entities)
 
-    devices.extend([NukiOpenerEntity(opener) for opener in bridge.openers])
+    platform = entity_platform.current_platform.get()
+    assert platform is not None
```

Comment:
error handling?

Reply:
As we discussed offline, it's not error handling but for type checking.

Reply:
There's no typing in the rest of the platform. Is it still needed?